cmd/mesh_bridge/main.go

```diff
@@ -0,0 +1,106 @@
+// +build !js
+
+// mesh_bridge is a short program that bridges two Mesh nodes. This is useful in cases where
+// we introduce a network-level breaking change but still want the liquidity from one network
+//to flow to another
+package main
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/0xProject/0x-mesh/rpc"
+	"github.com/0xProject/0x-mesh/zeroex"
+	"github.com/plaid/go-envvar/envvar"
+	log "github.com/sirupsen/logrus"
+)
+
+type clientEnvVars struct {
+	FirstWSRPCAddress  string `envvar:"FIRST_WS_RPC_ADDRESS"`
+	SecondWSRPCAddress string `envvar:"SECOND_WS_RPC_ADDRESS"`
+	Verbose            bool   `envvar:"VERBOSE"`
+}
+
+func main() {
+	env := clientEnvVars{}
+	if err := envvar.Parse(&env); err != nil {
+		panic(err)
+	}
+
+	if env.Verbose {
+		log.SetLevel(log.Level(4))
+	} else {
+		log.SetLevel(log.Level(0))
+	}
+
+	firstClient, err := rpc.NewClient(env.SecondWSRPCAddress)
+	if err != nil {
+		log.WithError(err).Fatal("could not create client")
+	}
+	stats, err := firstClient.GetStats()
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.WithField("stats", stats).Info("Spun up first client")
+
+	secondClient, err := rpc.NewClient(env.FirstWSRPCAddress)
+	if err != nil {
+		log.WithError(err).Fatal("could not create client")
+	}
+	stats, err = secondClient.GetStats()
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.WithField("stats", stats).Info("Spun up second client")
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		pipeOrders(secondClient, firstClient, "v8", "v9")
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		go pipeOrders(firstClient, secondClient, "v9", "v8")
+	}()
+
+	wg.Wait()
+}
+
+func pipeOrders(inClient, outClient *rpc.Client, inLabel, outLabel string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	orderEventsChan := make(chan []*zeroex.OrderEvent, 8000)
+	clientSubscription, err := inClient.SubscribeToOrders(ctx, orderEventsChan)
+	if err != nil {
+		log.WithError(err).Fatal("Couldn't set up OrderStream subscription")
+	}
+	defer clientSubscription.Unsubscribe()
+	signedOrdersCache := []*zeroex.SignedOrder{}
+	for {
+		select {
+		case orderEvents := <-orderEventsChan:
+			for _, orderEvent := range orderEvents {
+				if orderEvent.EndState != zeroex.ESOrderAdded {
+					continue
+				}
+				log.WithField("orderHash", orderEvent.OrderHash.Hex()).Info(fmt.Sprintf("Found new order %s -> %s", inLabel, outLabel))
+				signedOrdersCache = append(signedOrdersCache, orderEvent.SignedOrder)
+				// If more than 10 orders discovered, pipe them to the outClient
+				if len(signedOrdersCache) >= 10 {
```

Comment:
Nit: Let's make this magic number a constant.

Reply:
I think we need to rearchitect this or just remove it. As it's currently written, we could receive up to 9 orders and wait indefinitely to share them. This kind of delay is important and can have an impact on market makers.

The best way to do this is similar to [what we do in the `p2p` package when batching incoming messages](https://github.com/0xProject/0x-mesh/blob/1a4a775b4e6c6f726d004a1e897ffc88bc197180/p2p/node.go#L565). We wait for up to `maxReceiveBatch` messages or `receiveTimeout` seconds between messages, whichever comes first. 

(Though looking at the `p2p` code again I think there is a subtle bug where the data associated with a message could have been sent over the network but we canceled the `context` before processing it. Need to take a closer look here and maybe tweak it. Still the same principle applies.)

Reply:
Agreed. Using that type of logic now.